### DISCUSSION TITLE
error 500 doesn't let me push image

### DISCUSSION THREAD

**Discussion by AJUP86** [distribution/distribution#3731] [Q&A]
Hello, community ! 
Today i got into a problem that i haven't been able to solve.
After merge to test branch i deploy  this branch using 
docker-compose -f  test.yaml build 
docker-compose -f test.yaml push 
It builds it fine but at the moment of push it throws me the error without updating the image
my test.yaml push the images to a Docker Registry not to docker hub but i dont know what to do to fix this error and all documentations is available with docker command but not with docker compose. I have no experience with docker but i need to fix this. does anyone know what can be? the error is in the attachment
![docker error](https://user-images.githubusercontent.com/74145309/189427199-2d548fe9-c2e0-446a-b506-b795e4d77494.png)

**Comment by Jamstah** [comment]
Is there more information in the log from the registry?

**Reply by hyeyoung-leee** [reply]
Hi, I have the same issue.

The situation I am facing is as follows: 
We are using Swift as the backend storage, and when a specific layer is pushed, a 500 error occurs.
And in Swift, the size(Content-Length) of the corresponding object appears as 0.

Please help me to find the reason for this issue.

* Error log
```
time="2024-03-18T12:53:32.270355309Z" level=error msg="unknown error completing upload: swift: timeout expired while waiting for segments of /docker/registry/v2/blobs/sha256/b1/b1647e4debdbdf6f3b45e87505965596d73a6f9a82ab78ab3deca8423db13271/data to show up" auth.user.name="harbor_registry_user" go.version=go1.18.9 http.request.host=f8237f3a-kr1-registry.container.cloud.toast.com http.request.id=be95176e-d193-4e62-9023-9cf80e6b0bf3 http.request.method=PUT http.request.remoteaddr=10.77.11.242 http.request.uri="/v2/now/server-app/blobs/uploads/1a0a5f58-a52b-4dcb-b58d-3bb50dd414c8?_state=i_qTX_hqWjDk5-mSModBBGdPZge8V7dVJOv9fHeSL557Ik5hbWUiOiJub3cvc2VydmVyLWFwcCIsIlVVSUQiOiIxYTBhNWY1OC1hNTJiLTRkY2ItYjU4ZC0zYmI1MGRkNDE0YzgiLCJPZmZzZXQiOjEzOTkyMjI1LCJTdGFydGVkQXQiOiIyMDI0LTAzLTE4VDEyOjUzOjE3WiJ9&digest=sha256%3Ab1647e4debdbdf6f3b45e87505965596d73a6f9a82ab78ab3deca8423db13271" http.request.useragent="docker/24.0.6 go/go1.20.7 git-commit/1a79695 kernel/5.15.133.1-microsoft-standard-WSL2 os/linux arch/amd64 UpstreamClient(Docker-Client/24.0.6 \(windows\))" vars.name="now/server-app" vars.uuid=1a0a5f58-a52b-4dcb-b58d-3bb50dd414c8 
```
